sklearn/tree/_tree.pyx

```diff
@@ -918,10 +914,7 @@ cdef class Tree:
             raise ValueError("X.dtype should be np.float32, got %s" % X.dtype)
 
         # Extract input
-        cdef np.ndarray X_ndarray = X
-        cdef DTYPE_t* X_ptr = <DTYPE_t*> X_ndarray.data
-        cdef SIZE_t X_sample_stride = <SIZE_t> X.strides[0] / <SIZE_t> X.itemsize
-        cdef SIZE_t X_fx_stride = <SIZE_t> X.strides[1] / <SIZE_t> X.itemsize
+        cdef float [:, :] X_ndarray = X
```

Comment:
Shouldn't this be `cdef DTYPE_t` as well?

Reply:
yeah, I've already fixed that, took me a while to figure the difference between `np.float32` and `np.npy_float32` :P